File: lib/store.dart
import "product.dart";
import "package:mocktail/mocktail.dart";


class Store {
  final _inventory = Map<Symbol, int>();

  void addInventory(Symbol product, int amount) {
    _inventory[product] = amount;
  }

  bool hasEnough(Symbol product, int amount) {
    return _inventory[product]! >= amount;
  }
}

class MockStore extends Mock implements Store {}

File: lib/store.rb
class Store
  def initialize
    @inventory = {}
  end

  def addInventory(product, amount)
    @inventory[product] = amount
  end

  def removeFromInventory(product, amount)
    @inventory[product] -= amount
  end

  def getInventory(product)
    @inventory[product]
  end

  def enough?(product, amount)
    @inventory[product] >= amount
  end
end
